diff --strip-trailing-cr -urN libyaml.orig/yaml_private.h libyaml/yaml_private.h
--- libyaml.orig/yaml_private.h	2016-08-27 21:00:57.000000000 +0000
+++ libyaml/yaml_private.h	2017-07-31 10:05:54.000000000 +0000
@@ -5,6 +5,10 @@
 
 #include <yaml.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #include <assert.h>
 #include <limits.h>
 #include <stddef.h>
@@ -658,3 +662,6 @@
      (node).data.mapping.pairs.top = (node_pairs_start),                        \
      (node).data.mapping.style = (node_style))
 
+#ifdef __cplusplus
+}
+#endif
