diff --strip-trailing-cr -urN png.orig/pnglibconf.h png/pnglibconf.h
--- png.orig/pnglibconf.h	2017-09-29 08:40:57.000000000 +0000
+++ png/pnglibconf.h	2017-10-11 17:19:49.000000000 +0000
@@ -217,4 +217,14 @@
 #define PNG_sCAL_PRECISION 5
 #define PNG_sRGB_PROFILE_CHECKS 2
 /* end of settings */
+#undef PNG_CONSOLE_IO_SUPPORTED
+#undef PNG_CONVERT_tIME_SUPPORTED
+#ifdef PNG_NO_STDIO
+#  undef PNG_SIMPLIFIED_WRITE_STDIO_SUPPORTED
+#  undef PNG_STDIO_SUPPORTED
+#endif
+#if defined( __WATCOMC__ )
+#  undef PNG_API_RULE
+#  define PNG_API_RULE 2
+#endif
 #endif /* PNGLCONF_H */
