diff -urN expat.orig/xmlparse.c expat/xmlparse.c
--- expat.orig/xmlparse.c	2017-07-12 21:55:49.000000000 +0000
+++ expat/xmlparse.c	2017-07-30 20:44:40.000000000 +0000
@@ -790,7 +796,7 @@
 
   if (advapi32) {
     const RTLGENRANDOM_FUNC RtlGenRandom
-        = (RTLGENRANDOM_FUNC)GetProcAddress(advapi32, "SystemFunction036");
+        = (RTLGENRANDOM_FUNC)HB_WINAPI_GETPROCADDRESS(advapi32, "SystemFunction036");
     if (RtlGenRandom) {
       if (RtlGenRandom((PVOID)target, (ULONG)count) == TRUE) {
         success = 1;
@@ -808,10 +814,21 @@
 static unsigned long
 gather_time_entropy(void)
 {
-#ifdef _WIN32
+#ifdef _WINCE
+  SYSTEMTIME st;
+  GetSystemTime(&st);
+  return ( ( ( ( long ) st.wDay * 24 +
+                        st.wHour ) * 60 +
+                        st.wMinute ) * 60 +
+                        st.wSecond ) * 1000 + st.wMilliseconds;
+#elif defined(_WIN32)
   FILETIME ft;
   GetSystemTimeAsFileTime(&ft); /* never fails */
   return ft.dwHighDateTime ^ ft.dwLowDateTime;
+#elif defined(__WATCOMC__) && defined(__DOS__)
+  struct timeb tb;
+  ftime( &tb );
+  return tb.time ^ tb.millitm;
 #else
   struct timeval tv;
   int gettimeofday_res;
