diff --strip-trailing-cr -urN scrypt.orig/config.h scrypt/config.h
--- scrypt.orig/config.h	2017-02-11 07:29:20.000000000 +0000
+++ scrypt/config.h	2017-02-13 03:33:22.000000000 +0000
@@ -1,99 +1 @@
 /* config.h.in.  Generated from configure.ac by autoheader.  */
-
-/* Define to 1 if you have the `clock_gettime' function. */
-#undef HAVE_CLOCK_GETTIME
-
-/* Define to 1 if you have the <inttypes.h> header file. */
-#undef HAVE_INTTYPES_H
-
-/* Define to 1 if you have the <memory.h> header file. */
-#undef HAVE_MEMORY_H
-
-/* Define to 1 if you have the `mmap' function. */
-#undef HAVE_MMAP
-
-/* Define to 1 if you have the <openssl/aes.h> header file. */
-#undef HAVE_OPENSSL_AES_H
-
-/* Define to 1 if you have the `posix_memalign' function. */
-#undef HAVE_POSIX_MEMALIGN
-
-/* Define to 1 if you have the <stdint.h> header file. */
-#undef HAVE_STDINT_H
-
-/* Define to 1 if you have the <stdlib.h> header file. */
-#undef HAVE_STDLIB_H
-
-/* Define to 1 if you have the <strings.h> header file. */
-#undef HAVE_STRINGS_H
-
-/* Define to 1 if you have the <string.h> header file. */
-#undef HAVE_STRING_H
-
-/* Define to 1 if the system has the type `struct sysinfo'. */
-#undef HAVE_STRUCT_SYSINFO
-
-/* Define to 1 if `mem_unit' is a member of `struct sysinfo'. */
-#undef HAVE_STRUCT_SYSINFO_MEM_UNIT
-
-/* Define to 1 if `totalram' is a member of `struct sysinfo'. */
-#undef HAVE_STRUCT_SYSINFO_TOTALRAM
-
-/* Define to 1 if you have the `sysinfo' function. */
-#undef HAVE_SYSINFO
-
-/* Define to 1 if you have the <sys/param.h> header file. */
-#undef HAVE_SYS_PARAM_H
-
-/* Define to 1 if you have the <sys/stat.h> header file. */
-#undef HAVE_SYS_STAT_H
-
-/* Define to 1 if you have the <sys/sysctl.h> header file. */
-#undef HAVE_SYS_SYSCTL_H
-
-/* Define to 1 if you have the <sys/sysinfo.h> header file. */
-#undef HAVE_SYS_SYSINFO_H
-
-/* Define to 1 if you have the <sys/types.h> header file. */
-#undef HAVE_SYS_TYPES_H
-
-/* Define to 1 if you have the <unistd.h> header file. */
-#undef HAVE_UNISTD_H
-
-/* Name of package */
-#undef PACKAGE
-
-/* Define to the address where bug reports for this package should be sent. */
-#undef PACKAGE_BUGREPORT
-
-/* Define to the full name of this package. */
-#undef PACKAGE_NAME
-
-/* Define to the full name and version of this package. */
-#undef PACKAGE_STRING
-
-/* Define to the one symbol short name of this package. */
-#undef PACKAGE_TARNAME
-
-/* Define to the home page for this package. */
-#undef PACKAGE_URL
-
-/* Define to the version of this package. */
-#undef PACKAGE_VERSION
-
-/* Define to 1 if you have the ANSI C header files. */
-#undef STDC_HEADERS
-
-/* Version number of package */
-#undef VERSION
-
-/* Enable large inode numbers on Mac OS X 10.5.  */
-#ifndef _DARWIN_USE_64_BIT_INODE
-# define _DARWIN_USE_64_BIT_INODE 1
-#endif
-
-/* Number of bits in a file offset, on hosts where this is settable. */
-#undef _FILE_OFFSET_BITS
-
-/* Define for large files, on AIX-style hosts. */
-#undef _LARGE_FILES
diff --strip-trailing-cr -urN scrypt.orig/crypto_scrypt-ref.c scrypt/crypto_scrypt-ref.c
--- scrypt.orig/crypto_scrypt-ref.c	2017-02-11 07:29:13.000000000 +0000
+++ scrypt/crypto_scrypt-ref.c	2017-02-17 14:54:03.000000000 +0000
@@ -26,15 +26,11 @@
  * This file was originally written by Colin Percival as part of the Tarsnap
  * online backup system.
  */
-#include "scrypt_platform.h"
 
-#include <errno.h>
-#include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 
 #include "sha256.h"
-#include "sysendian.h"
 
 #include "crypto_scrypt.h"
 
@@ -223,19 +219,21 @@
 	size_t r = _r, p = _p;
 	uint32_t i;
 
+	int fail = -1;
+
 	/* Sanity-check parameters. */
 #if SIZE_MAX > UINT32_MAX
 	if (buflen > (((uint64_t)(1) << 32) - 1) * 32) {
-		errno = EFBIG;
+		fail = -2 /* EFBIG */;
 		goto err0;
 	}
 #endif
 	if ((uint64_t)(r) * (uint64_t)(p) >= (1 << 30)) {
-		errno = EFBIG;
+		fail = -2 /* EFBIG */;
 		goto err0;
 	}
 	if (((N & (N - 1)) != 0) || (N == 0)) {
-		errno = EINVAL;
+		fail = -3 /* EINVAL */;
 		goto err0;
 	}
 	if ((r > SIZE_MAX / 128 / p) ||
@@ -243,7 +241,7 @@
 	    (r > SIZE_MAX / 256) ||
 #endif
 	    (N > SIZE_MAX / 128 / r)) {
-		errno = ENOMEM;
+		fail = -4 /* ENOMEM */;
 		goto err0;
 	}
 
@@ -281,5 +279,5 @@
 	free(B);
 err0:
 	/* Failure! */
-	return (-1);
+	return (fail);
 }
diff --strip-trailing-cr -urN scrypt.orig/crypto_scrypt.h scrypt/crypto_scrypt.h
--- scrypt.orig/crypto_scrypt.h	2017-02-11 07:29:13.000000000 +0000
+++ scrypt/crypto_scrypt.h	2017-02-17 14:44:19.000000000 +0000
@@ -29,9 +29,6 @@
 #ifndef _CRYPTO_SCRYPT_H_
 #define _CRYPTO_SCRYPT_H_
 
-#include <stdint.h>
-#include <unistd.h>
-
 /**
  * crypto_scrypt(passwd, passwdlen, salt, saltlen, N, r, p, buf, buflen):
  * Compute scrypt(passwd[0 .. passwdlen - 1], salt[0 .. saltlen - 1], N, r,
diff --strip-trailing-cr -urN scrypt.orig/sha256.c scrypt/sha256.c
--- scrypt.orig/sha256.c	2017-02-11 07:29:13.000000000 +0000
+++ scrypt/sha256.c	2017-02-17 14:52:50.000000000 +0000
@@ -1,9 +1,7 @@
 #include <assert.h>
-#include <stdint.h>
 #include <string.h>
 
 #include "insecure_memzero.h"
-#include "sysendian.h"
 
 #include "sha256.h"
 
@@ -94,9 +92,9 @@
  * the 512-bit input block to produce a new state.
  */
 static void
-SHA256_Transform(uint32_t state[static restrict 8],
-    const uint8_t block[static restrict 64],
-    uint32_t W[static restrict 64], uint32_t S[static restrict 8])
+SHA256_Transform(uint32_t state[HB_C99_STATIC HB_C99_RESTRICT 8],
+    const uint8_t block[HB_C99_STATIC HB_C99_RESTRICT 64],
+    uint32_t W[HB_C99_STATIC HB_C99_RESTRICT 64], uint32_t S[HB_C99_STATIC HB_C99_RESTRICT 8])
 {
 	int i;
 
@@ -159,7 +157,7 @@
 
 /* Add padding and terminating bit-count. */
 static void
-SHA256_Pad(SHA256_CTX * ctx, uint32_t tmp32[static restrict 72])
+SHA256_Pad(SHA256_CTX * ctx, uint32_t tmp32[HB_C99_STATIC HB_C99_RESTRICT 72])
 {
 	size_t r;
 
@@ -213,7 +211,7 @@
  */
 static void
 _SHA256_Update(SHA256_CTX * ctx, const void * in, size_t len,
-    uint32_t tmp32[static restrict 72])
+    uint32_t tmp32[HB_C99_STATIC HB_C99_RESTRICT 72])
 {
 	uint32_t r;
 	const uint8_t * src = in;
@@ -271,7 +269,7 @@
  */
 static void
 _SHA256_Final(uint8_t digest[32], SHA256_CTX * ctx,
-    uint32_t tmp32[static restrict 72])
+    uint32_t tmp32[HB_C99_STATIC HB_C99_RESTRICT 72])
 {
 
 	/* Add padding. */
@@ -323,8 +321,8 @@
  */
 static void
 _HMAC_SHA256_Init(HMAC_SHA256_CTX * ctx, const void * _K, size_t Klen,
-    uint32_t tmp32[static restrict 72], uint8_t pad[static restrict 64],
-    uint8_t khash[static restrict 32])
+    uint32_t tmp32[HB_C99_STATIC HB_C99_RESTRICT 72], uint8_t pad[HB_C99_STATIC HB_C99_RESTRICT 64],
+    uint8_t khash[HB_C99_STATIC HB_C99_RESTRICT 32])
 {
 	const uint8_t * K = _K;
 	size_t i;
@@ -376,7 +374,7 @@
  */
 static void
 _HMAC_SHA256_Update(HMAC_SHA256_CTX * ctx, const void * in, size_t len,
-    uint32_t tmp32[static restrict 72])
+    uint32_t tmp32[HB_C99_STATIC HB_C99_RESTRICT 72])
 {
 
 	/* Feed data to the inner SHA256 operation. */
@@ -403,7 +401,7 @@
  */
 static void
 _HMAC_SHA256_Final(uint8_t digest[32], HMAC_SHA256_CTX * ctx,
-    uint32_t tmp32[static restrict 72], uint8_t ihash[static restrict 32])
+    uint32_t tmp32[HB_C99_STATIC HB_C99_RESTRICT 72], uint8_t ihash[HB_C99_STATIC HB_C99_RESTRICT 32])
 {
 
 	/* Finish the inner SHA256 operation. */
diff --strip-trailing-cr -urN scrypt.orig/sha256.h scrypt/sha256.h
--- scrypt.orig/sha256.h	2017-02-11 07:29:13.000000000 +0000
+++ scrypt/sha256.h	2017-02-17 14:52:16.000000000 +0000
@@ -2,7 +2,19 @@
 #define _SHA256_H_
 
 #include <stddef.h>
-#include <stdint.h>
+
+#include "hbdefs.h"
+#define uint8_t       HB_U8
+#define uint32_t      HB_U32
+#define uint64_t      HB_U64
+#define le32dec(x)    HB_GET_LE_UINT32(x)
+#define le32enc(x,y)  HB_PUT_LE_UINT32(x,y)
+#define le64dec(x)    HB_GET_LE_UINT64(x)
+#define le64enc(x,y)  HB_PUT_LE_UINT64(x,y)
+#define be32dec(x)    HB_GET_BE_UINT32(x)
+#define be32enc(x,y)  HB_PUT_BE_UINT32(x,y)
+#define be64dec(x)    HB_GET_BE_UINT64(x)
+#define be64enc(x,y)  HB_PUT_BE_UINT64(x,y)
 
 /*
  * Use #defines in order to avoid namespace collisions with anyone else's
