diff --strip-trailing-cr -urN bz2.orig/bzlib_pr.h bz2/bzlib_pr.h
--- bz2.orig/bzlib_pr.h	2010-09-10 22:41:55.000000000 +0000
+++ bz2/bzlib_pr.h	2017-01-04 00:40:28.000000000 +0000
@@ -85,7 +85,13 @@
 
 #else
 
+#ifdef __cplusplus
+extern "C" {
+#endif
 extern void bz_internal_error ( int errcode );
+#ifdef __cplusplus
+}
+#endif
 #define AssertH(cond,errcode) \
    { if (!(cond)) bz_internal_error ( errcode ); }
 #define AssertD(cond,msg)                do { } while (0)
