diff --strip-trailing-cr -urN liblzf.orig/lzf.h liblzf/lzf.h
--- liblzf.orig/lzf.h	2008-08-25 01:40:29.000000000 +0000
+++ liblzf/lzf.h	2017-03-16 12:35:59.000000000 +0000
@@ -37,6 +37,10 @@
 #ifndef LZF_H
 #define LZF_H
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /***********************************************************************
 **
 **	lzf -- an extremely fast/free compression/decompression-method
@@ -96,5 +100,9 @@
 lzf_decompress (const void *const in_data,  unsigned int in_len,
                 void             *out_data, unsigned int out_len);
 
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
 #endif
 
diff --strip-trailing-cr -urN liblzf.orig/lzf_c.c liblzf/lzf_c.c
--- liblzf.orig/lzf_c.c	2010-06-01 09:11:20.000000000 +0000
+++ liblzf/lzf_c.c	2017-03-16 12:35:59.000000000 +0000
@@ -120,7 +120,7 @@
    * special workaround for it.
    */
 #if defined (WIN32) && defined (_M_X64)
-  unsigned _int64 off; /* workaround for missing POSIX compliance */
+  unsigned __int64 off; /* workaround for missing POSIX compliance */
 #else
   unsigned long off;
 #endif
